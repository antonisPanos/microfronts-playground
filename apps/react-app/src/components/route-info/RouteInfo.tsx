import React, {Fragment, FunctionComponent, ReactElement} from 'react';
import {hot} from 'react-hot-loader';
import type {RouteComponentProps, WindowLocation} from '@reach/router';

export const RouteInfo: FunctionComponent<RouteComponentProps<{location: WindowLocation}>> = ({
  location
}): ReactElement => {
  return (
    <Fragment>
      <h3>ReactApp Child</h3>
      <hr />
      <pre>{JSON.stringify(location, null, 2)}</pre>
    </Fragment>
  );
};

declare let module: Record<string, unknown>;

export default hot(module)(RouteInfo);
