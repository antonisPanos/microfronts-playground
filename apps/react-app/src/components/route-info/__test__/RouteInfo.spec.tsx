import React from 'react';
import {shallow, ShallowWrapper} from 'enzyme';
import RouteInfo from '../RouteInfo';

describe('RouteInfo', () => {
  it('should render correctly', () => {
    const mainComp: ShallowWrapper = shallow(<RouteInfo />);

    expect(mainComp.debug()).toMatchSnapshot();
  });
});
