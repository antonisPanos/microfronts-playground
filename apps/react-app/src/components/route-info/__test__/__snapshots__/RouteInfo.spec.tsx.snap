// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteInfo should render correctly 1`] = `
"<Fragment>
  <h3>
    ReactApp Child
  </h3>
  <hr />
  <pre />
</Fragment>"
`;
